src/libsync/vfs/cfapi/hydrationjob.cpp

```diff
@@ -137,6 +160,70 @@ void OCC::HydrationJob::start()
     connect(_transferDataServer, &QLocalServer::newConnection, this, &HydrationJob::onNewConnection);
 }
 
+void OCC::HydrationJob::slotFolderIdError()
+{
+    // TODO: the following code is borrowed from PropagateDownloadEncrypted (see HydrationJob::onNewConnection() for explanation of next steps)
+    qCCritical(lcHydration) << "Failed to get encrypted metadata of folder" << _requestId << _localPath << _folderPath;
+    emitFinished(Error);
+}
+
+void OCC::HydrationJob::slotCheckFolderId(const QStringList &list)
+{
+    // TODO: the following code is borrowed from PropagateDownloadEncrypted (see HydrationJob::onNewConnection() for explanation of next steps)
```

Comment:
Is it possible to reuse the code without copying it?

Reply:
@FlexW Can be possible. Would need to factor it out from here and from `PropagateDownloadEncrypted ` and create a separate job.

Reply:
Okay. Would be great if we could avoid that duplication, but it might be too much for this pr.